engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/container/MultiInstanceBodyProcessor.java

```diff
@@ -115,12 +117,18 @@ public void onTerminate(
       final BpmnElementContext childContext) {
 
     final var bodyInstance = stateBehavior.getElementInstance(flowScopeContext);
-    final var loopCounter = bodyInstance.getMultiInstanceLoopCounter();
+
+    // on resolve incident, loop counter is potentially already set on the inner instance
+    final int loopCounter =
+        Optional.ofNullable(stateBehavior.getLocalVariable(childContext, LOOP_COUNTER_VARIABLE))
+            .map(MsgPackConverter::convertToJson)
+            .map(Integer::parseInt)
```

Comment:
I am not sure whether this is the best way to retrieve this variable

Reply:
This works nicely :+1: 

However, there is a small chance that it doesn't work. If the variable is modified (very unlikely but it is possible) then it would set the wrong loop variables. Or, it could create another incident or an error.

Instead, we could use the loop counter that was written to the state before.   

```
    final int loopCounter =
        Optional.ofNullable(stateBehavior.getElementInstance(childContext))
            .map(ElementInstance::getMultiInstanceLoopCounter)
            .orElse(bodyInstance.getMultiInstanceLoopCounter());
```

Reply:
That's a really nice solution, and I like it much better 👍

Reply:
True. We don't need the optional :+1: